added test script for python

havbon opened the pull request:
I have now taken all the tests, both json and not and put them in tests.json
Then i have disabled stderr testing for now, they are still present in json files for future.

I added the following features:

- run just a specific test case
- verify the exit code (to better test errors)

In order to run specific cases simply run test.py index,index,index (comma sep no spaces)
I planted a few issues for testing purposes.

closes #74

havbon commented:
I was going to say that i havent fully integrated the script into CI and removed the Perl script because i ran out of time, but then sonatype freaked out. I guess i will check it out tomorrow

bagder commented on tests.json:
```diff
@@ -0,0 +1,610 @@
+[
+    {
+        "stdin": "example.com",
```
This is not really `stdin`, I think maybe `cmdline` would be more appropriate? I would imagine that we actually want to eventually be able to provide data on stdin for tests.

havbon commented on test.py:
```diff
@@ -0,0 +1,144 @@
+        # returns false if the command does not contain keyword
+        if self.cmdline.find(stdinkeyword) == -1:
+            return False
+        output = subprocess.run(
```
@sonatype-lift ignore

havbon commented:
@sonatype-lift ignoreall

havbon commented:
after lots of tweaks and adjustments I have replaced the entire perl test suite to json, and switched the script to python. Checking stderr is somewhat implemented. I decided to replace the filter system, if `python.exe test.py --get` is run then it will only test the commands with `--get` in the arguments, this allows more flexible and faster testing

bagder commented:
I don't see the `--json` tests converted. Am I just blind?

havbon commented:
> I don't see the `--json` tests converted. Am I just blind?

Everything should be fixed now, I hope

bagder commented:
> Conflicting files test.pl 

Does this mean that there are a few recently added tests that are not converted over? I understand it is a little annoying but we have been adding tests to test.pl while you've worked on the python version and I'm just keen on making sure that we don't miss out on those. It will probably help if you don't remove the perl version in your PR so that we can make sure they both work equally fine before we remove the perl version in a separate PR.

havbon commented:
Okay i squashed and pushed, it should be fine now. I manually went through each test and I am quite certain I got every one, its also the same number of tests in perl and the json file.

bagder commented:
Thanks!
